### PR TITLE
Add extensions to FinetuningFileNotFoundError

### PR DESCRIPTION
**Pull request by b-chu**
Passes the list of supported extensions to the super class

### CODE REVIEW

**Review by milocress — APPROVED** [review]
🖖

**Review by jjanezhang — APPROVED** [review]
lgtm

File: llmfoundry/utils/exceptions.py [hunk 1]
```diff
@@ -496,6 +496,7 @@ def __init__(self, files_searched: list[str]) -> None:
         super().__init__(
             message,
             files_searched=files_searched,
+            extensions=SUPPORTED_EXTENSIONS,
```
Comment (milocress) [inline llmfoundry/utils/exceptions.py:499]:
```suggestion
    def __init__(self, files_searched: list[str], supported_extensions: Optional[list[str]]) -> None:
        if supported_extensions is None:
            from llmfoundry.data.finetuning.tasks import SUPPORTED_EXTENSIONS
            supported_extensions = SUPPORTED_EXTENSIONS
        message = (
            f'Could not find a file with any of ' + \
            f'the supported extensions: {supported_extensions}\n' + \
            f'at {files_searched}'
        )
        super().__init__(
            message,
            files_searched=files_searched,
            supported_extensions=supported_extensions,
        )
```

**Comment by milocress** [comment]
What odds do you want on a linter error I'll buy at 70 sell at 95

**Comment by milocress** [comment]
> What odds do you want on a linter error I'll buy at 70 sell at 95

Resolves to 1, no trade

**Pull request merged**